src/routes/Requests.tsx

```diff
@@ -0,0 +1,698 @@
+			requestCount++;
+		});
+		return gridFilters.customer
+			? o.customerKey == gridFilters.customer?.customerKey && requestCount > 0
+			: requestCount > 0;
+	});
+
+	const regionsQuery = useRegions();
+	const regions = regionsQuery?.data?.regions;
+
+	const locationsQuery = useLocations(gridFilters.region?.regionKey);
+	const locations = locationsQuery?.data?.locations;
+
+	const yardsQuery = useYards(gridFilters.region?.regionKey, gridFilters.region?.locationKey);
+	const yards = yardsQuery?.data?.yards;
+
+	// Sets filteredRequests by running setFilters() when lane changes or requests are refetched
+	useEffect(() => {
+		setFilteredRequests(setFilters(requests));
+	}, [gridFilters, requests]);
+
+	/** Checks if filters are set and return filtered request or all request
+	 * @param {RequestDisplay[]} r the returned requestQuery data object
+	 * @returns {RequestDisplay[]} filtered request data
+	 */
+	function setFilters(r: RequestDisplay[]) {
+		let requestFiltered = r;
+		const customerKey = gridFilters.customer?.customerKey;
+		const orderKey = gridFilters.order?.orderKey;
+		const regionKey = gridFilters.region?.regionKey;
+		const locationKey = gridFilters.location?.locationKey;
+		const yardKey = gridFilters.yard?.yardKey;
+		if (customerKey) {
+			requestFiltered = requestFiltered?.filter((rf) => {
+				return rf.order?.customer.customerKey == customerKey;
+			});
+		}
+		if (orderKey) {
+			requestFiltered = requestFiltered?.filter((rf) => {
+				return rf.order?.orderKey == orderKey;
+			});
+		}
+		if (regionKey) {
+			requestFiltered = requestFiltered?.filter((rf) => {
+				return rf.order?.regionKey == regionKey;
+			});
+		}
+		if (locationKey) {
+			requestFiltered = requestFiltered?.filter((rf) => {
+				return (
+					rf.requestHaulers?.find((rh) => {
+						return rh?.orderProduct?.location.locationKey == locationKey;
+					})?.requestKey == rf.requestKey
+				);
+			});
+		}
+		if (yardKey) {
+			requestFiltered = requestFiltered?.filter((rf) => {
+				return (
+					rf.requestHaulers?.find((rh) => {
+						return rh?.orderProduct?.yard.yardKey == yardKey;
+					})?.requestKey == rf.requestKey
+				);
+			});
+		}
+		return requestFiltered;
+	}
+
+	//Building the grid
+	const gridRef = useRef<AgGridReact>(null);
+
+	const gridOptions: GridOptions = {
+		// Default Grid Column Definitions
+		defaultColDef: {
+			sortable: true,
+			floatingFilter: false,
+			suppressMenu: true,
+			filter: "agTextColumnFilter",
+			resizable: true,
+		},
+		// Grid columns
+		columnDefs: [
+			{
+				//hidden Request Key column
+				field: "requestKey",
+				headerName: "Request Key",
+				filter: "agNumberColumnFilter",
+				hide: true,
+				pinned: true,
+				initialWidth: 100,
+				minWidth: 85,
+			},
+			{
+				field: "order.orderNumber",
+				headerName: "Order #",
+				initialWidth: 110,
+				minWidth: 110,
+			},
+			{
+				field: "order.customer.customerName",
+				headerName: "Customer",
+				initialWidth: 175,
+				minWidth: 120,
```

Comment:
I would also add flex widths to some columns to allow the grid to automatically use the extra space available.
![image](https://github.com/TAC-Insight/fw-scale-frontend/assets/4155618/2e7ecef3-1a23-4576-a0b5-b7f8f2da2390)

```suggestion
				minWidth: 120,
				flex: 1,
```

![image](https://github.com/TAC-Insight/fw-scale-frontend/assets/4155618/34f6deae-d465-4dba-9331-902db9f3e640)